subsys/net/buf.c

```diff
@@ -466,15 +469,17 @@ void net_buf_unref(struct net_buf *buf)
 		NET_BUF_DBG("buf %p ref %u pool_id %u frags %p", buf, buf->ref,
 			    buf->pool_id, buf->frags);
 
+		key = k_spin_lock(&pool->lock);
```

Comment:
Instead of spinlocks, could this be implemented similar way as how `net_pkt` does it, so using `atomic_cas()` with loop?
https://github.com/zephyrproject-rtos/zephyr/blob/961959fc5f2ba913a80af6e92ea493b373f90253/subsys/net/ip/net_pkt.c#L525-L544
Same comment to `net_buf_ref()`

Reply:
I thought so at first, but using `atomic_t` will increase the size of the `net_buf` structure.

Reply:
Good point, never mind then.